test/route_downcaser_test.rb

```diff
@@ -103,36 +162,75 @@ class RedirectTrueTests < ActiveSupport::TestCase
     end
   end
 
-
   class RedirectTrueExcludePatternsTests < ActiveSupport::TestCase
     setup do
       @app = MyMockApp.new
       RouteDowncaser.configuration do |config|
         config.redirect = true
         config.exclude_patterns = [/assets\//i, /fonts\//i]
+        config.include_patterns = nil
       end
     end
 
-    test "when redirect is true it does not redirect, if REQUEST_URI match exclude patterns" do
+    test "when redirect is true, it does not redirect if REQUEST_URI match exclude patterns" do
       callenv = { 'REQUEST_URI' => "fonts/Icons.woff", 'REQUEST_METHOD' => "GET" }
       RouteDowncaser::DowncaseRouteMiddleware.new(@app).call(callenv)
       assert_equal("fonts/Icons.woff", @app.env['REQUEST_URI'])
     end
 
-    test "when redirect is true it does not redirect, if PATH_INFO match exclude patterns" do
+    test "when redirect is true, it does not redirect if PATH_INFO match exclude patterns" do
       callenv = { 'PATH_INFO' => "fonts/Icons.woff", 'REQUEST_METHOD' => "GET" }
       RouteDowncaser::DowncaseRouteMiddleware.new(@app).call(callenv)
       assert_equal("fonts/Icons.woff", @app.env['PATH_INFO'])
     end
   end
 
+  class RedirectTrueIncludePatternsTests < ActiveSupport::TestCase
+    setup do
+      @app = MyMockApp.new
+      RouteDowncaser.configuration do |config|
+        config.redirect = true
+        config.exclude_patterns = nil
+        config.include_patterns = [/hello\//i]
+      end
+    end
+
+    test "when redirect is true, it does not redirect if REQUEST_URI doesn't match include patterns" do
```

Comment:
just from the perspective of someone attempting to understand the code, by reading the tests, it isn't very clear when read in plain english. I have some suggestions

`when redirect is true, it does not redirect if REQUEST_URI doesn't match include_patterns`
(this would indicate there is a method that the reader can go and look at to understand what 'include_patterns' is).

`when redirect is true, it does not redirect if REQUEST_URI does not match an included pattern`
(just reads better in plain english)

Also, from line 181, perhaps an improvement would be
`when redirect is true, it does not redirect if PATH_INFO matches an excluded pattern`
or
`when redirect is true, it does not redirect if PATH_INFO matches exclude_patterns`

While these comments are minor, such improvements helps speed up my understanding of the code, because the explanation of the test just makes total sense when read as a sentence :)